presto-hive/src/test/java/com/facebook/presto/hive/AbstractTestHiveClient.java

```diff
@@ -3380,10 +3409,13 @@ private void doInsert(HiveStorageFormat storageFormat, SchemaTableName tableName
 
             // "stage" insert data
             ConnectorInsertTableHandle insertTableHandle = metadata.beginInsert(session, tableHandle);
-            ConnectorPageSink sink = pageSinkProvider.createPageSink(transaction.getTransactionHandle(), session, insertTableHandle, PageSinkProperties.defaultProperties());
+            ConnectorPageSink sink = pageSinkProvider.createPageSink(transaction.getTransactionHandle(), session, insertTableHandle, pageSinkProperties);
             sink.appendPage(CREATE_TABLE_DATA.toPage());
             sink.appendPage(CREATE_TABLE_DATA.toPage());
             Collection<Slice> fragments = getFutureValue(sink.finish());
+            if (pageSinkProperties.isPartitionCommitRequired()) {
+                metadata.commitPartitions(session, insertTableHandle, fragments);
+            }
```

Comment:
Let's also do similar tests for `doInsertIntoNewPartition`, `testInsertIntoExistingPartition`